Tasks/ServiceFabricComposeDeploy/task.json

```diff
@@ -0,0 +1,162 @@
+            "type": "pickList",
+            "label": "Repository Credentials",
+            "defaultValue": "Endpoint",
+            "required": true,
+            "options": {
+                "Endpoint": "Endpoint",
+                "UsernamePassword": "Username and Password",
+                "None": "None"
+            },
+            "groupname": "repository",
+            "helpMarkDown": "Choose if/how credentials for the docker registry will be provided."
+        },
+        {
+            "name": "dockerRegistryEndpointName",
+            "type": "connectedService:dockerregistry",
+            "label": "Docker Registry Connection",
+            "required": false,
+            "helpMarkDown": "Select a Docker registry connection. If a certificate matching the Server Certificate Thumbprint in the Cluster Connection is installed, it will be used to encrypt the password; otherwise the password will not be encrypted.",
+            "groupname": "repository",
+            "visibleRule": "repositoryCredentials = Endpoint"
+        },
+        {
+            "name": "repositoryUserName",
+            "type": "string",
+            "label": "Repository User Name",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "Username for the Docker registry",
+            "groupname": "repository",
+            "visibleRule": "repositoryCredentials = UsernamePassword"
+        },
+        {
+            "name": "repositoryPassword",
+            "type": "string",
+            "label": "Repository Password",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "Password for the Docker registry. If the password is not encrypted, it is recomended that you use a secret variable to store it.",
+            "groupname": "repository",
+            "visibleRule": "repositoryCredentials = UsernamePassword"
+        },
+        {
+            "name": "passwordEncrypted",
+            "type": "boolean",
+            "label": "Password Encrypted",
+            "defaultValue": true,
+            "required": false,
+            "helpMarkDown": "It is recomended to encrypt your password using [Invoke-ServiceFabricEncryptText](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-application-secret-management#encrypt-application-secrets). If you do not, and a certificate matching the Server Certificate Thumbprint in the Cluster Connection is installed, it will be used to encrypt the password; otherwise the password will not be encrypted.",
+            "groupname": "repository",
+            "visibleRule": "repositoryCredentials = UsernamePassword"
+        },
+        {
+            "name": "deployTimeoutSec",
+            "type": "string",
+            "label": "Deploy Timeout (s)",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "Timeout in seconds for deploying the application.",
```

Comment:
The cmdlets don't give any info on that, I'm also hesitant for us to document something, as the underlying cmdlet could change it's default.